test/shared_type_tester.dart

```diff
@@ -0,0 +1,148 @@
+@TestOn('browser')
+@JS()
+library react_test_utils_test;
+
+import 'dart:async';
+import 'dart:html' as html;
+
+import 'package:js/js.dart';
+import 'package:test/test.dart';
+
+sharedTypeTests(Function testTypeValue,
```

Comment:
This typing should be more specific:

```suggestion
typedef void TestTypeValue<T>(T value);
void sharedTypeTests(TestTypeValue testTypeValue,
```

Reply:
We should avoid unnecessary `typedef`s since we're Dart 2 only now... we can just use the new inline `Function` syntax.

Reply:
Good point; the only reason I put in the typedef was because I wasn't sure if the semantics were the same for generics

Reply:
Addressed in https://github.com/kealjones-wk/react-dart/pull/3